util/src/main/java/com/google/cloud/hadoop/util/RetryHttpInitializer.java

```diff
@@ -95,6 +96,7 @@ public void initialize(HttpRequest request) throws IOException {
       headers.setUserAgent(options.getDefaultUserAgent());
     }
     headers.putAll(options.getHttpHeaders());
+    request.setInterceptor(new InvocationIdInterceptor(request.getInterceptor()));
```

Comment:
It is not obvious why we are passing one interceptor to another. Is it because this will override the existing interceptor? If yes, please add a comment since it is not very obvious. Also we use ChainingInterceptors at other places for similar scenarios.

Reply:
Nothing is getting overridden. All interceptors are getting chained via this. Unlike gRPC in HTTP there is no way to provide more than one interceptor for request and this feels like the only way via which we can have multiple interceptors(one after another) act on a single request.

`ChainingInterceptors` which you are referring  to is not a interceptor but an `initializer:ChainingHttpRequestInitializer`  which chains multiple `initializers` and `interceptors` using the same exact logic which I used (or other way round).

Reply:
yes, I was referring about ChainingInterceptors. Please do add a comment explaining that this is doing a chaining interceptor. Also why not make use of existing chaining interceptor?

Reply:
There is NO `ChainingInterceptors`  there is `ChainingHttpRequestInitializer`. I am not creating another `HTTPInitializer` just an interceptor and hence there is no way(or no need to) I can use `ChainingHttpRequestInitializer`. `Interceptor` and `Initializer` are two different things.
IIUC, If someone is creating a new interceptor and they can add it to any of the `HTTPInitializer` used.